options/common.go

```diff
@@ -0,0 +1,114 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/gluster/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/store"
+)
+
+const (
+	clusterOptsPrefix string = store.GlusterPrefix + "clusteropts/"
+	volumeOptsPrefix  string = store.GlusterPrefix + "volumeopts/"
+)
+
+// GetClusterOption gets cluster configuration from etcd store,
+// returns default value if not available in store. Returns error if
+// not available in defaults list.
+func GetClusterOption(key string) (interface{}, error) {
+	var v interface{}
+
+	// Get Value from Store
+	clusterKey := clusterOptsPrefix + key
+	resp, e := gdctx.Store.Get(context.TODO(), clusterKey)
+	if e == nil {
+		if resp.Count == 1 {
+			if e = json.Unmarshal(resp.Kvs[0].Value, &v); e != nil {
+				return nil, e
+			}
+			return v, nil
+		}
+	}
+
+	// If not available get value from defaults
+	dv, dvOk := clusterDefaultOptions[key]
+	if dvOk {
+		return dv.defaultValue, nil
+	}
+
+	// Else return error
+	return "", errors.New("Invalid Option")
```

Comment:
Please use `nil` instead of empty string as return type is `interface{}`. Also in `GetVolumeOption()`